src/pipelinedefinition.cpp

```diff
@@ -492,33 +492,34 @@ class NodeValidator {
             return StatusCode::PIPELINE_NOT_ENOUGH_SHAPE_DIMENSIONS_TO_DEMULTIPLY;
         }
         if (demultiplicatorNodeInfo.demultiplyCount.value() != 0) {
+            // TODO -1 vs 0 as demultiplexing value
             if (shape[0] != 0) {
```

Comment:
Shape should never be 0 to begin with. Shape from custom node should be translated to -1